application/src/main/kotlin/no/nav/amt/tiltak/application/ControllerAdvice.kt

```diff
@@ -0,0 +1,123 @@
+package no.nav.amt.tiltak.application
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import no.nav.amt.tiltak.core.exceptions.EndringsmeldingIkkeAktivException
+import no.nav.amt.tiltak.core.exceptions.UnauthorizedException
+import no.nav.amt.tiltak.core.exceptions.ValidationException
+import org.apache.commons.lang3.exception.ExceptionUtils
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.ResponseStatus
+
+@ControllerAdvice
+open class ControllerAdvice(
+	@Value("\${rest.include-stacktrace: false}") private val includeStacktrace: Boolean
+) {
+
+	private val logger = LoggerFactory.getLogger(javaClass)
+
+	@ResponseStatus(HttpStatus.NOT_FOUND)
+	@ExceptionHandler(NoSuchElementException::class)
+	fun handleNotFoundException(e: NoSuchElementException): ResponseEntity<Response> {
+		logger.info(e.message, e)
+
+		return buildResponse(
+			status = HttpStatus.NOT_FOUND,
+			exception = e
+		)
+	}
+
+	@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+	@ExceptionHandler(IllegalStateException::class)
+	fun handleIllegalStateException(e: IllegalStateException): ResponseEntity<Response> {
+		logger.error(e.message, e)
+
+		return buildResponse(
+			status = HttpStatus.INTERNAL_SERVER_ERROR,
+			exception = e
+		)
+	}
+
+	@ResponseStatus(HttpStatus.UNAUTHORIZED)
+	@ExceptionHandler(UnauthorizedException::class)
+	fun handleUnauthorizedException(e: UnauthorizedException): ResponseEntity<Response> {
+		logger.info(e.message, e)
+
+		return buildResponse(
+			status = HttpStatus.UNAUTHORIZED,
+			exception = e
+		)
+	}
```

Comment:
Dette skal egentlig være HttpStatus.FORBIDDEN.

Kan være greit å lage en UnauthenticatedException også for HttpStatus.UNAUTHORIZED